src/sc_crawler/vendors/aws.py

```diff
@@ -748,13 +792,177 @@ def inventory_server_prices_spot(vendor):
+    "standard": {
+        "maxIopsvolume": 200,
+        "maxThroughputvolume": 90,
+        "minVolumeSize": 1 / 1024,
+        "maxVolumeSize": 1,
+    },
+    "gp2": {
+        "maxIopsvolume": 16_000,
+        "maxThroughputvolume": 250,
+        "minVolumeSize": 1 / 1024,
+        "maxVolumeSize": 16,
+    },
+    "gp3": {
+        "maxIopsvolume": 16_000,
+        "maxThroughputvolume": 250,
+        "minVolumeSize": 1 / 1024,
+        "maxVolumeSize": 16,
+    },
+    "st1": {
+        "maxIopsvolume": 500,
+        "maxThroughputvolume": 500,
+        "minVolumeSize": 125 / 1024,
+        "maxVolumeSize": 16,
+    },
+    "sc1": {
+        "maxIopsvolume": 250,
+        "maxThroughputvolume": 250,
+        "minVolumeSize": 125 / 1024,
+        "maxVolumeSize": 16,
+    },
+}
+
+
+def inventory_storages(vendor):
+    vendor.progress_tracker.start_task(
+        name="Searching for Storages", n=len(storage_manual_data)
+    )
+    # look up all volume types in us-east-1
+    products = []
+    for volume_type in storage_types:
+        products.extend(
+            _boto_get_products(
+                service_code="AmazonEC2",
+                filters={
+                    "volumeType": volume_type,
+                    "location": "US East (N. Virginia)",
+                },
+            )
+        )
+        vendor.progress_tracker.advance_task()
+    vendor.progress_tracker.hide_task()
+
+    for product in products:
+        attributes = product["product"]["attributes"]
+        product_id = attributes["volumeApiName"]
+
+        def get_attr(key: str) -> float:
+            return extract_last_number(
+                attributes.get(
+                    key,
+                    storage_manual_data[product_id][key],
+                )
+            )
+
+        storage_type = (
+            StorageType.HDD if "HDD" in attributes["storageMedia"] else StorageType.SSD
+        )
+        Storage(
+            id=product_id,
+            vendor=vendor,
+            name=attributes["volumeType"],
+            description=attributes["storageMedia"],
+            storage_type=storage_type,
+            max_iops=get_attr("maxIopsvolume"),
+            max_throughput=get_attr("maxThroughputvolume"),
+            min_size=get_attr("minVolumeSize") * 1024,
+            max_size=get_attr("maxVolumeSize") * 1024,
+        )
+
+    vendor.log(f"{len(products)} Storages synced.")
+
+
 def inventory_storage_prices(vendor):
-    pass
+    loc2dc = _location_datacenter_map(vendor)
+    vendor_storages = {x.id: x for x in vendor.storages}
+    vendor.progress_tracker.start_task(
+        name="Searching for Storage Prices", n=len(storage_manual_data)
+    )
+    # look up all volume types in us-east-1
+    products = []
+    for volume_type in storage_types:
+        products.extend(
+            _boto_get_products(
+                service_code="AmazonEC2",
+                filters={"volumeType": volume_type},
+            )
+        )
```

Comment:
@bra-fsn this is actually a part that we could parallelize easily as not interacting with the database. do you have a preference on how to do this? e.g. ~effortless update to this for/extend loop to make the API requests in parallel.

Reply:
https://docs.python.org/3/library/concurrent.futures.html is the easiest, but I'm not sure it worths it. Does it take long to collect these?
Check this as well: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html#multithreading-or-multiprocessing-with-clients

Reply:
Awesome, thanks! It takes a couple of seconds for each volume type in this exact example, but there are some other cases when looking up products in 30+ regions each taking 5-15 seconds, so it will be more useful there.

Reply:
@bra-fsn if this pattern looks OK, I will replicate to some similar cases -- please let me know :pray: https://github.com/SpareCores/sc-crawler/pull/9/commits/9042a8f54161a3696081f2223af1cbdcda721576#diff-e6e0a018c8f44553e71647fed43533c8ee2188d909487b5c167317926288a129R824-R827